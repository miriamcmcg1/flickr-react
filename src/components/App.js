import React from "react";
import { HashRouter as Router, Route } from "react-router-dom";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faStroopwafel } from "@fortawesome/free-solid-svg-icons";
import MainPage from "./MainPage";
import Image from "./Image";

library.add(faStroopwafel);

function App() {
  return (
    <div className="App">
      <MuiThemeProvider>
        <Router>
          <div>
            <Route exact path="/" component={MainPage} />
            <Route
              name="image"
              path="/image/:id/:server/:farm/:secret/:title"
              component={Image}
            />
          </div>
        </Router>
      </MuiThemeProvider>
    </div>
  );
}

export default App;
