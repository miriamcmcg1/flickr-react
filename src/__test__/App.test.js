import React from "react";
import { shallow } from "enzyme";
import App from "../components/App";
import "../setupTests";

describe("Render App", () => {
  describe('render', () => {
      test('should render App', () => {
        expect(shallow(<App />)).toMatchSnapshot();
      });
  });
});
