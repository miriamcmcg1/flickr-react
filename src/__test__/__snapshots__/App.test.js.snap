// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App render should render App 1`] = `
<div
  className="App"
>
  <MuiThemeProvider>
    <HashRouter>
      <div>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          name="image"
          path="/image/:id/:server/:farm/:secret/:title"
        />
      </div>
    </HashRouter>
  </MuiThemeProvider>
</div>
`;
